### PR TITLE
Report OMP threads in PairHMM (resolves #56)

### LINKED ISSUE #56: Report OMP threads in PairHMM
Add code to report:
* Number of OMP threads requested 
* Number of OMP threads available
* Number of OMP threads used

If number of threads used is less than the number requested, generate a warning message.

### PR DESCRIPTION
**Pull request by erniebrau**
Resolves #56

### CODE REVIEW

File: src/main/native/pairhmm/IntelPairHmm.cc [hunk 1]
```diff
@@ -37,7 +37,22 @@ JNIEXPORT void JNICALL Java_com_intel_gkl_pairhmm_IntelPairHmm_initNative
   g_use_double = use_double;
   
 #ifdef _OPENMP
-  g_max_threads = std::min((int)max_threads, omp_get_max_threads());
+  int avail_threads = omp_get_max_threads();
+  int req_threads = (int)max_threads;
```
Comment (gspowley) [inline src/main/native/pairhmm/IntelPairHmm.cc]: Do you need the typecast on max_threads? We needed it in `std::min`, not sure if we need it here.

File: src/main/native/pairhmm/IntelPairHmm.cc [hunk 2]
```diff
@@ -37,7 +37,22 @@ JNIEXPORT void JNICALL Java_com_intel_gkl_pairhmm_IntelPairHmm_initNative
...
+  INFO("Requested threads: %d", req_threads);
+  if (req_threads > avail_threads) {
+    WARN("Using %d available threads, but %d were requested", g_max_threads, req_threads);
+  }
+  else {
+    INFO("Using %d threads", g_max_threads);
+  }
+#else
+  if ((int)max_threads != 1) {
+    WARN("Ignoring request for %d threads; not using OpenMP implementation", (int)max_threads);
```
Comment (gspowley) [inline src/main/native/pairhmm/IntelPairHmm.cc]: Same typecast comment on these lines.

**Comment by erniebrau** [comment]
@gspowley -- you were right, they casts were not necessary. I removed them. Let me know if there's anything else, and if I can go ahead and merge.

**Comment by gspowley** [comment]
Looks good, you can merge.

**Pull request merged**